internal/services/redis/redis_cache_resource.go

```diff
@@ -817,15 +817,13 @@ func expandRedisConfiguration(d *pluginsdk.ResourceData) (*redis.RedisCommonProp
 	}
 
 	// RDB Backup
-	// nolint : staticcheck
-	v, valExists := d.GetOkExists("redis_configuration.0.rdb_backup_enabled")
-	if valExists {
-		if v := raw["rdb_backup_enabled"].(bool); v {
-			if connStr := raw["rdb_storage_connection_string"].(string); connStr == "" {
-				return nil, fmt.Errorf("The rdb_storage_connection_string property must be set when rdb_backup_enabled is true")
-			}
+	config := d.GetRawConfig().AsValueMap()["redis_configuration"].AsValueSlice()[0].AsValueMap()
```

Comment:
We can't be using `GetRawConfig` since this functionality won't be available when we move to framework.

There is a wrapper function that can be used instead `pluginsdk.IsExplicitlyNullInConfig`.

Reply:
`rdb_storage_connection_string` is nested in `redis_configuration`.   Cannot get it with `pluginsdk.IsExplicitlyNullInConfig(d, "redis_configuration.0.rdb_backup_enabled")`

Reply:
In this case there's no need to use either `IsExplicitlyNullInConfig` or `d.GetRawConfig().AsValueMap()` - this should only be set if the value is set to `true`?

Reply:
`rdb_backup_enabled` can be `false` or `true`.  I think we can filter it by the SKU type.

Reply:
@xuzhang3 I used the `Basic` SKU and it was failing, saying that the `rdb_backup_enabled` option is only allowed for `Premium`, so I assume it will be set to true regardless of the inputs of SKU